### PR TITLE
GIPHY - Independent publisher connector

### PR DESCRIPTION
**Pull request by priyanshu870**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by priyanshu870** [comment]
image

**Comment by priyanshu870** [comment]
image

**Comment by priyanshu870** [comment]
image
This is how the Get-GIF action looks like

**Comment by Amjed-Ayoub** [comment]
Hi @priyanshu870, May I ask you please to sign our CLA so we can review your pull request.

Please let me know if you have any questions.

Thank you so much

**Comment by priyanshu870** [comment]
@microsoft-github-policy-service agree company="no company specified"

**Comment by priyanshu870** [comment]
@microsoft-github-policy-service agree company="no company specified"

**Comment by priyanshu870** [comment]
Done @Amjed-Ayoub !
Can you please review the pull request now and tell me how to fix the errors I'm getting

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,848 @@
...
+        },
+        "summary": "Get GIF",
+        "description": "Returns a list of GIFs related to the search string provided as input",
+        "operationId": "Get-GIF",
+        "parameters": [
+          {
+            "name": "API_KEY",
+            "in": "query",
+            "required": true,
+            "type": "string"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:795]: @priyanshu870 please add 
"description": "Your description", could be "Enter your API Key."
"x-ms-summary": "Your Summary", could be "APIKey"

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,848 @@
...
+          },
+          {
+            "name": "q",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "Search String"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:802]: @priyanshu870 please add 
"x-ms-summary": "Your Summary"

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,848 @@
...
+          },
+          {
+            "name": "limit",
+            "in": "query",
+            "required": false,
+            "type": "integer"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:808]: @priyanshu870 please add 
"description": "Your description",
"x-ms-summary": "Your Summary"

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,848 @@
...
+          },
+          {
+            "name": "offset",
+            "in": "query",
+            "required": false,
+            "type": "integer"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:814]: @priyanshu870 please add 
"description": "Your description",
"x-ms-summary": "Your Summary"

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,848 @@
...
+          },
+          {
+            "name": "rating",
+            "in": "query",
+            "required": false,
+            "type": "string"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:820]: @priyanshu870 please add 
"description": "Your description",
"x-ms-summary": "Your Summary"

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,848 @@
...
+          },
+          {
+            "name": "lang",
+            "in": "query",
+            "required": false,
+            "type": "string"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:826]: @priyanshu870 please add 
"description": "Your description",
"x-ms-summary": "Your Summary"

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,848 @@
...
+  "schemes": [
+    "https"
+  ],
+  "consumes": [],
+  "produces": [],
+  "paths": {
+    "/": {
+      "get": {
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:19]: @priyanshu870 default responses are not allowed. Please define a proper response. please see our docs [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses) and [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,848 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "GIPHY",
+    "description": "This custom connector wraps around GIPHY API. It can be used to get GIFs related to a search string that is provided as input",
+    "version": "1.0"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:6]: @priyanshu870 please add contact section.
 "contact": {
      "name": "Full name",
      "url": "Contact URL",
      "email": "Contact email address"
    }
Please see out template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiDefinition.swagger.json)

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,848 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "GIPHY",
+    "description": "This custom connector wraps around GIPHY API. It can be used to get GIFs related to a search string that is provided as input",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:5]: @priyanshu870 Please remove the API word from the description and add a period at the end of it.

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,848 @@
...
+      "name": "API_KEY"
+    }
+  },
+  "security": [
+    {
+      "API Key": []
+    }
+  ],
+  "tags": []
+}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:848]: @priyanshu870 please add "x-ms-connector-metadata" section.
 "x-ms-connector-metadata": [
    {
      "propertyName": "Website",
      "propertyValue": ""
    },
    {
      "propertyName": "Privacy policy",
      "propertyValue": ""
    },
    {
      "propertyName": "Categories",
      "propertyValue": ""
    }
  ]

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json (hunk 1 above)
Comment (priyanshu870) [reply independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json]: Done

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json (hunk 3 above)
Comment (priyanshu870) [reply independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json]: Done

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json (hunk 4 above)
Comment (priyanshu870) [reply independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json]: Done

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json (hunk 5 above)
Comment (priyanshu870) [reply independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json]: Done

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json (hunk 6 above)
Comment (priyanshu870) [reply independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json]: Done

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json (hunk 7 above)
Comment (priyanshu870) [reply independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json]: Done

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json (hunk 8 above)
Comment (priyanshu870) [reply independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json]: Done

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiProperties.json [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+          "tooltip": "Provide your API_KEY",
+          "constraints": {
+            "tabIndex": 2,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#007ee5",
```
[the remark this answers is not in the document]
Comment (priyanshu870) [reply independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiProperties.json]: Done

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiProperties.json [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+    "capabilities": [],
+    "publisher": "Priyanshu Srivastav"
```
[the remark this answers is not in the document]
Comment (priyanshu870) [reply independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiProperties.json]: Done

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 13]
```diff
@@ -0,0 +1,878 @@
...
+"x-ms-connector-metadata": [
+{
+  "propertyName": "Website",
+  "propertyValue": ""
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:868]: @priyanshu870 this should be a valid url.

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 14]
```diff
@@ -0,0 +1,878 @@
...
+},
+{
+  "propertyName": "Privacy policy",
+  "propertyValue": ""
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:872]: @priyanshu870 this should be a valid url.

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 15]
```diff
@@ -0,0 +1,878 @@
...
+},
+{
+  "propertyName": "Categories",
+  "propertyValue": ""
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:876]: @priyanshu870 this is not a valid category. you can choose one of the below.
Available categories are: [AI, Business Management, Business Intelligence, Collaboration, Commerce, Communication, Content and Files, Finance, Data, Human Resources, Internet of Things, IT Operations, Lifestyle and Entertainment, Marketing, Productivity, Sales and CRM, Security, Social Media, Website]

File: independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json [hunk 16]
```diff
@@ -0,0 +1,878 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "GIPHY",
+    "description": "This custom connector wraps around GIPHY. It can be used to get GIFs related to a search string that is provided as input.",
+    "version": "1.0",
+    "contact": {
+    "name": "Priyanshu Srivastav",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Proposal - GIPHY (Independent Publisher)/apiDefinition.swagger.json:8]: @priyanshu870 please fix the indentation. all files should be properly formatted.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by Amjed-Ayoub** [comment]
Hello @priyanshu870,
I hope you are doing well.
I am following up with you to see how things are going with preparing your connector.
Please let us know if you have any questions.
Thank you.

**Comment by Amjed-Ayoub** [comment]
Hello @priyanshu870,
I hope you are doing well.
I am following up in my messages below to see if you need any help with updating your connector artifacts.
Please do not hesitate to reach out to me if you have any questions as we are always here to help. 😊
I look forward to hearing from you.
Thank you very much.

**Comment by priyanshu870** [comment]
Hey @Amjed-Ayoub!
Sorry for being inactive in the last couple of weeks. Was busy with my college exams.
I made the necessary changes in the files as requested. Can you please look into this and suggest what more chenges are needed

**Comment by Amjed-Ayoub** [comment]
Hi @priyanshu870 thank you so much for your hard work on this.
May I ask you please to remove the "Proposal" from the folder name as well as "(Independent Publisher)". please just keep the connector name "GIPHY".
Thank you so much.

**Comment by priyanshu870** [comment]
@Amjed-Ayoub 
Renamed the folder

**Comment by Amjed-Ayoub** [comment]
@priyanshu870 Please fix your files. now you have two folders :-)
image

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @priyanshu870,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by priyanshu870** [comment]
Thankyou so much @Amjed-Ayoub for your help!
Looking forward to it!😍

Just a question -
When will the certificate of Independent Publisher connector be provided to me? Will it be after the connector is fully deployed (3-4 weeks time) or is there something else that needs to be done.
Thank you

**Comment by Amjed-Ayoub** [comment]
> Thankyou so much @Amjed-Ayoub for your help! Looking forward to it!😍
> 
> Just a question - When will the certificate of Independent Publisher connector be provided to me? Will it be after the connector is fully deployed (3-4 weeks time) or is there something else that needs to be done. Thank you

Hi @priyanshu870,

Adding @jopanchal to help with this, as I am not sure about this.

Thank you so much.

**Comment by Amjed-Ayoub** [comment]
Hello @priyanshu870,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @priyanshu870,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Review by Amjed-Ayoub — APPROVED** [review]

**Pull request merged**